src/utilities/git.jl

```diff
@@ -1,5 +1,86 @@
-git_checkout(branch::AbstractString) = run(`git checkout $(branch)`)
+const COMPATHELPER_GIT_COMMITTER_NAME = "CompatHelper Julia"
+const COMPATHELPER_GIT_COMMITTER_EMAIL = "compathelper_noreply@julialang.org"
 
-function git_clone(url::AbstractString, local_path::AbstractString)
-    return run(`git clone $(url) $(local_path)`)
+function get_git_name_and_email(; env=ENV)
+    name = if haskey(env, "GIT_COMMITTER_NAME")
+        env["GIT_COMMITTER_NAME"]
+    else
+        COMPATHELPER_GIT_COMMITTER_NAME
+    end
+
+    email = if haskey(env, "GIT_COMMITTER_EMAIL")
+        env["GIT_COMMITTER_EMAIL"]
+    else
+        COMPATHELPER_GIT_COMMITTER_EMAIL
+    end
+
+    return name, email
+end
+
+git_checkout(branch::AbstractString) = run(`git checkout $branch`)
+
+function git_add(; items::AbstractString=".", flags::AbstractString="")
+    all_items = split(items, " "; keepempty=false)
+    all_flags = split(flags, " "; keepempty=false)
+    x = run(`git add $all_flags $all_items`)
+    return nothing
+end
+
+function git_push(
+    remote::AbstractString,
+    branch::AbstractString,
+    pkey_filename::Union{AbstractString,Nothing}=nothing;
+    force=false,
+    env=ENV,
+)
+force_flag = force ? ["-f"] : []
+    name, email = get_git_name_and_email(; env=env)
+
+    withenv(
+        "GIT_SSH_COMMAND" => isnothing(pkey_filename) ? "ssh" : "ssh -i $pkey_filename"
+    ) do
+        # TODO: Do we also want to set `committer.name` and `committer.email`?
```

Comment:
Any opinions on this?

Reply:
I briefly looked this up, and apparently you need one or both of them set.

Reply:
Figured I'd add both here just because